cms/service/ProxyService.py

```diff
@@ -562,28 +562,31 @@ def dataset_updated(self, task_id):
         """
         with SessionGen() as session:
             task = Task.get_from_id(task_id, session)
-            dataset = task.active_dataset
-
-            # This ProxyService may focus on a different contest, and it should
-            # ignore this update.
-            if task.contest_id != self.contest_id:
-                logger.debug("Ignoring dataset change for task %d of contest "
-                             "%d (this ProxyService considers contest %d "
-                             "only).", task_id, task.contest.id,
-                             self.contest_id)
-                return
-
-            logger.info("Dataset update for task %d (dataset now is %d).",
-                        task.id, dataset.id)
-
-            # max_score and/or extra_headers might have changed.
-            self.reinitialize()
-
-            for submission in task.submissions:
-                # Update RWS.
-                if not submission.participation.hidden and \
-                        submission.official and \
-                        submission.get_result() is not None and \
-                        submission.get_result().scored():
-                    for operation in self.operations_for_score(submission):
-                        self.enqueue(operation)
+            if task is not None:
```

Comment:
It might be more readable to invert the check here (`if task is None`), returning early in that case. It would avoid having to indent the rest of the code.

Reply:
You are right it would make it more readable. I think I fixed it and pushed, I am not sure it's the first time I open a pull request, sorry for any github-wise misbehaviour :D